sdk/test-utils/test-utils/src/tracing/chaiAzureTrace.ts

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { OperationTracingOptions, useInstrumenter } from "@azure/core-tracing";
+import { assert } from "chai";
+import { MockInstrumenter } from "./mockInstrumenter";
+import { SpanGraph, SpanGraphNode } from "./spanGraphModel";
+
+/**
+ * Augments Chai with support for Azure Tracing functionality
+ *
+ * Sample usage:
+ *
+ * ```ts
+ * import chai from "chai";
+ * import { chaiAzureTrace } from "@azure/test-utils";
+ * chai.use(chaiAzureTrace);
+ *
+ * it("supportsTracing", async () => {
+ *   await assert.supportsTracing((updatedOptions) => myClient.doSomething(updatedOptions), ["myClient.doSomething"]);
+ * });
+ * ```
+ * @param chai - The Chai instance
+ */
+function chaiAzureTrace(chai: Chai.ChaiStatic): void {
+  // expect(() => {}).to.supportsTracing() syntax
+  chai.Assertion.addMethod("supportTracing", function<T>(
+    this: Chai.AssertionStatic,
+    expectedSpanNames: string[],
+    options?: T
+  ) {
+    return assert.supportsTracing(this._obj, expectedSpanNames, options, this._obj);
+  });
+
+  // assert.supportsTracing(() => {}) syntax
+  chai.assert.supportsTracing = supportsTracing;
+}
+
+const instrumenter = new MockInstrumenter();
+/**
+ * The supports Tracing function does the verification of whether the core-tracing is supported correctly with the client method
+ * This function verifies the root span, if all the correct spans are called as expected and if they are closed.
+ * @param callback - Callback function of the client that should be invoked
+ * @param expectedSpanNames - List of span names that are expected to be generated
+ * @param options - Options for either Core HTTP operations or custom options for the callback
+ * @param thisArg - optional this parameter for the callback
+ */
+async function supportsTracing<
+  Options extends { tracingOptions?: OperationTracingOptions },
+  Callback extends (options: Options) => Promise<unknown>
+>(
+  callback: Callback,
+  expectedSpanNames: string[],
+  options?: Options,
+  thisArg?: ThisParameterType<Callback>
+) {
+  useInstrumenter(instrumenter);
```

Comment:
Should we just spy on the tracingClient instead and not test core functionality repeatedly?